packages/config-resolver/src/endpointsConfig/resolveEndpointsConfig.ts

```diff
@@ -57,13 +57,17 @@ export interface EndpointsResolvedConfig extends Required<EndpointsInputConfig>
 
 export const resolveEndpointsConfig = <T>(
   input: T & EndpointsInputConfig & PreviouslyResolved
-): T & EndpointsResolvedConfig => ({
-  ...input,
-  tls: input.tls ?? true,
-  endpoint: input.endpoint
-    ? normalizeEndpoint({ ...input, endpoint: input.endpoint })
-    : () => getEndpointFromRegion(input),
-  isCustomEndpoint: input.endpoint ? true : false,
-  useDualstackEndpoint: normalizeBoolean(input.useDualstackEndpoint!),
-  useFipsEndpoint: normalizeBoolean(input.useFipsEndpoint!),
-});
+): T & EndpointsResolvedConfig => {
+  const useDualstackEndpoint = normalizeBoolean(input.useDualstackEndpoint!);
+  const useFipsEndpoint = normalizeBoolean(input.useFipsEndpoint!);
+  return {
```

Comment:
Why adding the `return` instead of arrow function?

Reply:
The params `useDualstackEndpoint` and `useFipsEndpoint` need to be passed at two places:
* `getEndpointFromRegion`
* populate in Output

Since they're computed in advance, we have to add function body.
We still use an arrow function though.